source/Private/Get-LocalizedDataRecursive.ps1

```diff
@@ -88,7 +88,7 @@ function Get-LocalizedDataRecursive
             else
             {
                 # Assuming derived class that is not part of this module.
-                throw 'The class ''{0}'' is not part of module DscResource.Base and no BaseDirectory was passed. Please provide BaseDirectory.'
+                throw ($script:localizedData.ThrowClassIsNotPartOfModule -f $name)
```

Comment:
Also reading the docs, should this not use one of the exceptions in `DscResource.Common`, e.g. `New-InvalidDataException`?

Reply:
Maybe. 🤔  Those `New-*Exception` is used to throw error messages in the resources, but when I worked in SqlServerDsc to create public commands those old methods are no longer appropriate, and instead started to write up new suggestions https://github.com/dsccommunity/SqlServerDsc/blob/main/CONTRIBUTING.md#commands. Maybe the same can be used for private commands instead of the old `New-*Exception`? See example here that throws a terminating error: https://github.com/dsccommunity/SqlServerDsc/blob/main/source/Private/Assert-Feature.ps1  🤔 

Either way, I think maybe open issue so that change that can be another PR?